### PR TITLE
Add support for some crypto functions

### PR DESCRIPTION
**Pull request by lyrixx**

### CODE REVIEW

**Comment by Wirone** [comment]
*crypto 😅

**Comment by lyrixx** [comment]
Oupsy. Thanks and fixed

**Comment by tigitz** [comment]
Hey guys,

I have my own implementation and wanted to share some thoughts.

Why not just use `encrypt` and `decrypt` instead of `_with_password`? Are you planning to support other encryption methods that justify this distinction?

In my case, I often encrypt files (e.g., .env, auth.json, certificates), so I use a dedicated `encrypt_file`($filePath, $outputPath) method more often than a basic string encryption helper, which I call `encrypt_string`($string) for clarity.

My implementation is based on [Illuminate’s Encrypter](https://github.com/illuminate/encryption/blob/master/Encrypter.php) which might be helpful to look into for inspiration. For example, `#[\SensitiveParameter]` could be a useful additionn here IMO. 

Just sharing some ideas, take or leave whatever you find useful 😉

**Comment by lyrixx** [comment]
PR is ready

File: src/functions.php [hunk 1]
```diff
@@ -826,6 +826,48 @@ function yaml_dump(mixed $input, int $inline = 2, int $indent = 4, int $flags =
...
+    }
+
+    $content = file_get_contents($sourcePath);
+    if (false === $content) {
+        throw new \RuntimeException(\sprintf('Failed to read the file "%s".', $sourcePath));
+    }
+
+    $encrypted = encrypt_with_password($content, $password);
+
+    Container::get()->fs->dumpFile($destinationPath ?? $sourcePath, $encrypted);
```
[the remark this answers is not in the document]
Comment (tigitz) [reply src/functions.php:852]: Or perhaps prompting for confirmation before overwriting could be a solution?

[the remark this answers is not in the document]
Comment (lyrixx) [reply src/functions.php]: done ✅

**Review by pyrech — APPROVED** [review]
LGTM

**Comment by lyrixx** [comment]
@tigitz Is it okay for you?

**Comment by tigitz** [comment]
Yup, aside from possibly considering preservation of file permissions when transitioning between `file_get_contents `and `file_put_contents`, everything looks great to me. 👍

**Comment by lyrixx** [comment]
Here we go! All commentes has been addressed. Everything should be okay!

**Review by pyrech — APPROVED** [review]
Awesome, thanks for this feature

**Pull request merged**